src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Utility.php

```diff
@@ -148,4 +153,26 @@ private function loadInactiveCustomerQuote()
 
         return $quote;
     }
+
+    /**
+     * Creates a new quote for customerId and sets it to active
+     *
+     * @param int $customerId
+     * @return Mage_Sales_Model_Quote
+     */
+    private function createNewCustomerQuote($customerId)
```

Comment:
This logic is only partially correct, please see how a new quote gets created for a customer and how customer properties get assigned to the quote: ```Shopgate_Cloudapi_Model_Api2_Carts_Rest_Customer_V2```
Since you will be reusing the functionality, you can move that logic to the utility class and use by both classes.